From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Mon, 1 Jul 2019 10:38:02 +0200
Subject: window-x11: Don't double-check for unmanaging windows

When looking for the best fallback focus window, we ignore it if it is in the
unmanaging state, but meta_stack_get_default_focus_window() does this is check
for us already.

So, ignore the redundant test.

https://gitlab.gnome.org/GNOME/mutter/merge_requests/669

(cherry picked from commit 9aee47daa984e5cc4a1ccefb3aacfbb2729cfbc1)

Origin: https://gitlab.gnome.org/GNOME/mutter/merge_requests/669
Bug-Ubuntu: https://launchpad.net/bugs/1791574
Bug-Ubuntu: https://launchpad.net/bugs/1834583
Bug-GNOME: https://gitlab.gnome.org/GNOME/mutter/issues/308
Bug-GNOME: https://gitlab.gnome.org/GNOME/mutter/issues/660
Bug-JetBrains: https://youtrack.jetbrains.com/issue/IDEA-198187
Applied-Upstream: 3.28.5
---
 src/x11/window-x11.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/x11/window-x11.c b/src/x11/window-x11.c
index 29ea3a6..c09ba3e 100644
--- a/src/x11/window-x11.c
+++ b/src/x11/window-x11.c
@@ -852,9 +852,6 @@ meta_window_x11_focus (MetaWindow *window,
                       if (!focus_window)
                         break;
 
-                      if (focus_window->unmanaging)
-                        continue;
-
                       if (focus_window->input)
                         break;
 
